modules/rostests/apitests/umkm/Exceptions.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Tests for exception handling
+ * COPYRIGHT:   Copyright 2024 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#include "precomp.h"
+#include <apitest.h>
+#include <windows.h>
+#include <stdio.h>
+#include <ndk/rtlfuncs.h>
+
+void
+RunHandleRaiseExceptionTest(void)
+{
+    _SEH2_TRY
+    {
+       RaiseException(0xDEADBEEF, 0, 0, NULL);
+       exit(-1);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        /* This should be executed */
+        exit(1);
+    }
+    _SEH2_END;
+
+    /* This should not be executed */
+    exit(-2);
+}
+
+void
+RunContinueRaiseExceptionTest(void)
+{
+    _SEH2_TRY
+    {
+       RaiseException(0xDEADBEEF, 0, 0, NULL);
+       exit(1);
+    }
+    _SEH2_EXCEPT(EXCEPTION_CONTINUE_EXECUTION)
+    {
+        exit(-1);
+    }
+    _SEH2_END;
+    exit(-2);
+}
+
+void
+RunHandleNoncontinuableExceptionTest(void)
+{
+    _SEH2_TRY
+    {
+       RaiseException(0xDEADBEEF, EXCEPTION_NONCONTINUABLE, 0, NULL);
+       exit(-1);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        exit(1);
+    }
+    _SEH2_END;
+    exit(-2);
+}
+
+void
+RunContinueNoncontinuableExceptionTest(void)
+{
+    int HandlerExecuted = 0;
+    ULONG ExceptionCode = 0;
+
+    _SEH2_TRY
+    {
+        _SEH2_TRY
+        {
+            RaiseException(0xDEADBEEF, EXCEPTION_NONCONTINUABLE, 0, NULL);
+            exit(-1);
+        }
+        _SEH2_EXCEPT(EXCEPTION_CONTINUE_EXECUTION)
+        {
+            exit(-2);
+        }
+        _SEH2_END;
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        HandlerExecuted = 1;
+        ExceptionCode = _SEH2_GetExceptionCode();
+    }
+    _SEH2_END;
+
+    ok(HandlerExecuted, "Exception handler not executed\n");
+    ok(ExceptionCode == EXCEPTION_NONCONTINUABLE_EXCEPTION, "Wrong exception code: 0x%lx\n", ExceptionCode);
+}
+
+void
+RunFilterFastfailTest(void)
+{
+    _SEH2_TRY
+    {
+        __fastfail(1);
+        exit(-1);
+    }
+    _SEH2_EXCEPT(exit(-2), EXCEPTION_EXECUTE_HANDLER)
+    {
+        exit(-3);
+    }
+    _SEH2_END;
+    exit(-4);
+}
+
+static
+int
+RunSubtest(PSTR SubtestName)
+{
+    CHAR szCmdLine[MAX_PATH];
+
+    GetModuleFileNameA(NULL, szCmdLine, _countof(szCmdLine));
+    lstrcatA(szCmdLine, " Exceptions ");
+    lstrcatA(szCmdLine, SubtestName);
+
+    // Run the test
+    return system(szCmdLine);
```

Comment:
Usually `CreateProcess` + `winetest_wait_child_process` is used for this, but that would mean changing the exit code of your process (it uses anything non-zero and below 256 as error count, and 256 or above as exception)